assets/src/blocks/Articles/ArticlesList.js

```diff
@@ -1,108 +1,17 @@
-import { Component } from '@wordpress/element';
 import { ArticlePreview } from './ArticlePreview';
 
-const { apiFetch } = wp;
-const { addQueryArgs } = wp.url;
-
-export class ArticlesList extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      posts: props.posts || []
-    };
-
-    this.loadArticles = this.loadArticles.bind(this);
-  }
-
-  componentDidMount() {
-    this.loadArticles();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { article_count, tags, posts, post_types, ignore_categories, page } = this.props;
-    if (
-      article_count !== prevProps.article_count ||
-      tags.length !== prevProps.tags.length ||
-      post_types.length !== prevProps.post_types.length ||
-      ignore_categories !== prevProps.ignore_categories ||
-      posts.length !== prevProps.posts.length
-    ) {
-      this.loadArticles();
-    } else if (page !== prevProps.page) {
-      this.loadArticles(page);
-    }
-  }
-
-  async loadArticles(page) {
-    const {
-      article_count,
-      posts,
-      tags,
-      post_types,
-      ignore_categories,
-      postType,
-      postId,
-      setTotalPages
-    } = this.props;
-
-    const args = {
-      article_count: article_count && article_count > 0 ? article_count : 0,
-      post_types,
-      posts,
-      tags,
-      ignore_categories
-    };
-
-    if (page) {
-      args.offset = page * article_count;
-    }
-
-    if (postType === 'post') {
-      args.exclude_post_id = postId;
-    }
-
-    const queryArgs = {
-      path: addQueryArgs('/planet4/v1/get-posts', args)
-    };
-
-    try {
-      const result = await apiFetch(queryArgs);
-
-      let newPosts = [];
-      let totalPages = 0;
-      if (result) {
-        if (page) {
-          newPosts = [...this.state.posts, ...result.recent_posts];
-        } else {
-          newPosts = result.recent_posts;
-          totalPages = result.total_pages;
-        }
-      }
-      this.setState({
-        posts: newPosts
-      });
-      if (totalPages && setTotalPages) {
-        setTotalPages(totalPages);
-      }
-    } catch (error) {
-      console.log(error);
-    }
-  }
-
-  render() {
-    const { posts } = this.state;
-    const { postType } = this.props;
-
-    return (
-      <div className="article-list-section clearfix">
-        {posts && posts.length > 0 && posts.map(post =>
-          <ArticlePreview
-            key={post.post_title}
-            isCampaign={postType === 'campaign'}
-            post={post}
-          />
-        )}
-      </div>
-    );
-  }
+export const ArticlesList = (props) => {
+  const { posts, postType } = props;
```

Comment:
Instead of passing down `posts` as props, couldn't we use `useArticlesFetch` to get them here too?

Reply:
Hmm, short story: we could, but not sure if we should :)

long story: Initially I had set it up so that `useArticlesFetch` returned a function to reset the posts, for when the attributes change, intended only for the editor. That's why I couldn't call this function here. But then I removed that function as it's a lot easier to handle this inside `useArticlesFetch`.

So after that change it would in fact work the same way if we call `useArticlesFetch` here. However there is a high chance that we want to use some of the state in `ArticlesEditor`: for example when they change the number of articles shown, we actually know how many posts there are in total. So we could display that currently there are a total of `x` posts matched by the parameters of the block.